org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/codegen/AnnotationTargetTypeConstants.java

```diff
@@ -19,6 +19,8 @@
 
 public interface AnnotationTargetTypeConstants {
 
+	int NonJVMS4_7_20_TargetLocation         = -1;
```

Comment:
I was a bit at loss, where in JVMS exactly I should look for an explanation...

Reply:
> I was a bit at loss, where in JVMS exactly I should look for an explanation...

Per above, I need a sentinel value for the targetType parameter of org.eclipse.jdt.internal.compiler.ast.Annotation.isRuntimeTypeInvisible(int) method that is outside the legal range of values for type annotation target locations enumerated by JVMS Table 4.7.20  - 

Better suggestions welcome

Reply:
Rather than a lot of hand waving (where I couldn't be sure whether or not I was talking bullshit), I experimented with disentangling how we handle type annotations and declaration annotations as much as possible. Please find my result in PR #1117 - which is created on top of this PR but pushed separately as not to interfere with your work here.

The key connection to this constant is: the invocations where the new constant was used could be _removed_ altogether :)

WDYT?

Reply:
> I was a bit at loss, where in JVMS exactly I should look for an explanation...

I have in the latest additional push, gotten rid of this head-scratch producing enumeration.

Reply:
> Rather than a lot of hand waving (where I couldn't be sure whether or not I was talking bullshit), I experimented with disentangling how we handle type annotations and declaration annotations as much as possible. Please find my result in PR #1117 - which is created on top of this PR but pushed separately as not to interfere with your work here.
> 
> The key connection to this constant is: the invocations where the new constant was used could be _removed_ altogether :)
> 
> WDYT?

I will review this - I suggest let us do this in a follow up ticket - (I'll raise one)